### ISSUE TITLE
Limit number of cores used

### ISSUE THREAD

**Issue by Wytamma** [microsoft/CameraTraps#286]
I'm using the batch detector on a HPC and need to limit the number of cores used to the number requested. If I used the `--cores` the batch detector creates a pool of workers which uses a lot more memory and does not support logging or checkpointing. Is there a way to limit the number of cores used without creating a workerpool? I think it's related to [set_inter_op_parallelism_threads](https://www.tensorflow.org/api_docs/python/tf/config/threading/set_inter_op_parallelism_threads), but I'm not sure how to set tensorflow config from `run_tf_detector_batch.py`.

**Comment by agentmorris** [comment]
BTW adding checkpoints for parallelized jobs (ncores > 1) wouldn't be that big a deal, but I don't think that would actually help you here; as I understand it, your issue is that you want to *limit* the number of cores used, so I think the issue of checkpointing isn't related.

**Comment by Wytamma** [comment]
Hi @agentmorris! Thanks for the reply. Yep the HPC is running Linux. Your solution kinda works for me. As far as I can tell `tf.ConfigProto(...)` and `tf.config.threading...` don't work. However setting the env vars has some effect. It seems that the number of cpu used is limit to around 1.5 * the value of `OMP_NUM_THREADS`. I've adjusted my code to set `OMP_NUM_THREADS` etc. To the number of cpus requested // 2. Thanks again!

**Comment by Wytamma** [comment]
Actually... with this configuration the number of cores used never goes above requested amount... config and env vars are controlling different settings?
```
os.environ["OMP_NUM_THREADS"] = str(requested_cores)
os.environ["TF_NUM_INTRAOP_THREADS"] = str(requested_cores)
os.environ["TF_NUM_INTEROP_THREADS"] = str(requested_cores)

config = tf.ConfigProto(intra_op_parallelism_threads=1,
                        inter_op_parallelism_threads=1, 
                        allow_soft_placement=True,
                        device_count = {'CPU': 1})
```

**Comment by agentmorris** [comment]
Great, glad we found a solution.  

For posterity, can you confirm that you are in fact passing that config object when you create the session?  I.e.:

```
self.tf_session = tf.Session(graph=detection_graph,config=config)
```

**Comment by Wytamma** [comment]
Yep without passing the config object it uses ~1.5 * OMP_NUM_THREADS cores

**Comment by agentmorris** [comment]
Great, closing this issue.
